### ISSUE TITLE
Scaled-Dot-Product-Attention Has Broadcastable Mask

### ISSUE THREAD

**Issue by YifanShenSZ** [pytorch/pytorch#120810]
### 📚 The doc issue

The [scaled-dot-product-attention doc](https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html) claims
```
attn_mask (optional Tensor) – Attention mask; shape (N,...,L,S). 
```
However, if we try the attached [broadcast-mask.py.zip](https://github.com/pytorch/pytorch/files/14438879/broadcast-mask.py.zip), we will see
```
query shape = torch.Size([3, 2, 4, 7, 16])
key shape = torch.Size([3, 2, 4, 5, 16])
value shape = torch.Size([3, 2, 4, 5, 32])
mask shape = torch.Size([1, 1, 5])
attention shape = torch.Size([3, 2, 4, 7, 32])
```
This means our real shape requirement for `attn_mask` is `(..., S)` and broadcastable to `(N, ..., L, S)`

### Suggest a potential alternative/fix

Please update the [scaled-dot-product-attention doc](https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html) so it is consistent with the code behavior.

Or, if the code behavior is to-be-changed to conform to doc, please either raise exception in code or mark deprecation warning.

cc @svekars @brycebortree @jbschlosser @bhosmer @cpuhrsch @erichan1 @drisspg @mikaylagawarecki
